packages/cli/src/commands/setup/deploy/providers/render.js

```diff
@@ -67,28 +69,37 @@ export const prismaDataSourceCheck = (database) => {
       content: RENDER_YAML(''),
     }
   }
-  const content = fs.readFileSync(getPaths().api.dbSchema).toString()
-  const detectedDatabase = content.match(
-    /(?<=datasource DS.*\n\W*provider\W*)\w+/
-  )
-  if (detectedDatabase == database) {
-    switch (database) {
-      case 'postgresql':
-        return {
-          path: path.join(getPaths().base, 'render.yaml'),
-          content: RENDER_YAML(POSTGRES_YAML),
-        }
-      case 'sqlite':
-        return {
-          path: path.join(getPaths().base, 'render.yaml'),
-          content: RENDER_YAML(SQLITE_YAML),
-        }
-      default:
-        throw new Error(`
+  if (!fs.existsSync('api/db/schema.prisma')) {
+    throw new Error("Could not find prisma schema at 'api/db/schema.prisma'")
+  }
+
+  const getPrismaConfig = async () => {
+    const schema = await getSchema('api/db/schema.prisma')
+    const config = await getConfig({ datamodel: schema })
+    return config
+  }
+
+  getPrismaConfig().then((prismaConfig) => {
```

Comment:
@dac09 I'm having issues getting all the promise, async error handling to work in this section.

Reply:
Might be worth using async/await instead.

```
const prismaConfig = await getPrismaConfig()


 const detectedDatabase = prismaConfig.datasources[0].activeProvider

// Note === operator
  if (detectedDatabase === database) {
   // xx
  } else {
    throw new Error(`
    Prisma datasource provider is detected to be ${detectedDatabase}.
    Option 1: Update your schema.prisma provider to be ${database}, then run
    yarn rw prisma migrate dev
    yarn rw setup deploy render --database ${database}
    Option 2: Rerun setup deloy command with current schema.prisma provider:
    yarn rw setup deploy render --database ${detectedDatabase}`)
  }
```

Reply:
And you'll have to return the function called here https://github.com/redwoodjs/redwood/pull/2099/files#r615134675 for the error to propogate through I think!